test/UrlHelperTest.php

```diff
@@ -128,14 +127,29 @@ public function testProvidedParametersOverrideAnyPresentInARouteResultWhenGenera
         $result->getMatchedRouteName()->willReturn('resource');
         $result->getMatchedParams()->willReturn(['id' => 1]);
 
-        $this->router->generateUri('resource', ['id' => 2])->willReturn('URL');
+        $this->router->generateUri('resource', ['id' => 2], [])->willReturn('URL');
 
         $helper = $this->createHelper();
         $helper->setRouteResult($result->reveal());
 
         $this->assertEquals('URL', $helper('resource', ['id' => 2]));
     }
 
+    public function testWillNotReuseRouteResultParamsIfReuseResultParamsFlagIsFalseWhenGeneratingUri()
+    {
+        $result = $this->prophesize(RouteResult::class);
+        $result->isFailure()->willReturn(false);
+        $result->getMatchedRouteName()->willReturn('resource');
+        $result->getMatchedParams()->willReturn(['id' => 1]);
+
+        $this->router->generateUri('resource', [], [])->willReturn('URL');
+
+        $helper = $this->createHelper();
+        $helper->setRouteResult($result->reveal());
+
+        $this->assertEquals('URL', $helper('resource', [], [], null, ['reuse_result_params' => false]));
```

Comment:
I was concerned that the `reuse_result_params` flag was available previously and/or was a flag argument to the helper, but just verified that the current master branch has no awareness of this. As such, good to go.